sql/hive/v0.12.0/src/main/scala/org/apache/spark/sql/hive/Shim12.scala

```diff
@@ -241,8 +241,14 @@ private[hive] object HiveShim {
       Decimal(hdoi.getPrimitiveJavaObject(data).bigDecimalValue())
     }
   }
+
+  implicit def prepareWritable(shimW: ShimWritable): Writable = {
+    shimW.writable
+  }
 }
 
+case class ShimWritable(writable: Writable)
```

Comment:
If we skip `ShimWriteable`, we then need to remove `implicit` from `prepareWriteable` and explicitly call it to do the fixing. Is it better? If so, I can do it in this way.

It does not break Hive 12 because we just pass the underlying writable object without touching it. We only do the fixing on Hive 13.